modules/server/src/main/scala/higherkindness/compendium/db/queries/Queries.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.compendium.db.queries
+
+import doobie._
+import doobie.implicits.toSqlInterpolator
+
+object Queries {
+
+  def checkIfExistsQ(id: String): Query0[Boolean] =
+    sql"""
+          SELECT exists (SELECT true FROM protocols WHERE id=$id)
+       """.query[Boolean]
+
+  def upsertProtocolIdQ(id: String): Update0 =
```

Comment:
Not fully sure about `ON CONFLICT DO NOTHING`, but would it mean that if the record already exists we are not updating it? If that's the case, `insertProtocolIdQ ` would be more accurate?

Reply:
Since `Protocol` metadata is now really simple (only storing protocol `id`), upsert operation is like insert operation. But it will contain more data in the future, which will have a different `ON CONFLICT` logic.